src/python/pants/backend/python/docs/sphinx/target_types.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import os
+from pathlib import PurePath
+from typing import Sequence
+
+from pants.core.goals.package import OutputPathField
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    InvalidFieldException,
+    MultipleSourcesField,
+    Target,
+)
+from pants.util.strutil import softwrap
+
+
+class SphinxProjectSourcesField(MultipleSourcesField):
+    # TODO: support markdown
+    default = ("conf.py", "**/*.rst")
+    expected_file_extensions = (".py", ".rst")
```

Comment:
This is a bad setting because you may want to include a `static` or `template` folder - I'll remove it on a followup w/ dedicated tests.